image-rs/src/layer_store.rs

```diff
@@ -0,0 +1,79 @@
+use std::fs;
+use std::path::PathBuf;
+use std::sync::atomic::{AtomicUsize, Ordering};
+
+/// The LayerStore abstracts the image layers storage on the file system.
+pub struct LayerStore {
+    /// OCI image layer data store dir.
+    pub data_dir: PathBuf,
+
+    /// Next layer index
+    layers_index: AtomicUsize,
+}
+
+impl LayerStore {
+    /// Retrieves the next layer index by computing the maximum index from the
+    /// existing layers in data dir.
+    pub fn get_layer_index(data_dir: PathBuf) -> AtomicUsize {
+        let mut next = 0;
+        if data_dir.exists() {
+            if let Ok(paths) = fs::read_dir(data_dir) {
+                for entry_path in paths.filter_map(|result| result.ok()) {
```

Comment:
I understand we are ignoring [IO errors](https://doc.rust-lang.org/beta/std/fs/struct.ReadDir.html#errors) here, is this intentional?

Reply:
Added logging.

Reply:
I changed to propagated the error